components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/validators/LogoutRequestValidator.java

```diff
@@ -0,0 +1,184 @@
+            if (log.isDebugEnabled()) {
+                log.debug(notification);
+            }
+            samlMessageContext.setValidStatus(false);
+            String errorResponse = SAMLLogoutUtil.buildErrorResponse(samlMessageContext, logoutRequest.getID(),
+                    SSOConstants.StatusCodes.REQUESTOR_ERROR, notification);
+            throw new SAMLLogoutException(notification, errorResponse, logoutRequest.getDestination(),
+                    samlMessageContext.getRelayState());
+        }
+        if (StringUtils.isBlank(logoutRequest.getIssuer().getFormat()) ||
+                !(ISSUER_FORMAT.equals(logoutRequest.getIssuer().getFormat()))) {
+            String notification = "Invalid Issuer Format in the logout request";
+            if (log.isDebugEnabled()) {
+                log.debug(notification);
+            }
+            samlMessageContext.setValidStatus(false);
+            String errorResponse = SAMLLogoutUtil.buildErrorResponse(samlMessageContext, logoutRequest.getID(),
+                    SSOConstants.StatusCodes.REQUESTOR_ERROR, notification);
+            throw new SAMLLogoutException(notification, errorResponse, logoutRequest.getDestination(),
+                    samlMessageContext.getRelayState());
+        }
+    }
+
+    /**
+     * Validate the subject of the logout request.
+     *
+     * @param logoutRequest {@link LogoutRequest} object to be validated.
+     * @throws SAMLLogoutException If Subject of the Logout Request validation fails.
+     */
+    private void isSubjectValid(LogoutRequest logoutRequest) throws SAMLLogoutException {
+
+        if (logoutRequest.getNameID() == null && logoutRequest.getBaseID() == null
+                && logoutRequest.getEncryptedID() == null) {
+            String notification = "Subject Name should be specified in the Logout Request";
+            if (log.isDebugEnabled()) {
+                log.debug(notification);
+            }
+            samlMessageContext.setValidStatus(false);
+            String errorResponse = SAMLLogoutUtil.buildErrorResponse(samlMessageContext, logoutRequest.getID(),
+                    SSOConstants.StatusCodes.REQUESTOR_ERROR, notification);
+            throw new SAMLLogoutException(notification, errorResponse, logoutRequest.getDestination(),
+                    samlMessageContext.getRelayState());
+        }
+    }
+
+    /**
+     * Validate the signature of the logout request.
+     *
+     * @param logoutRequest {@link LogoutRequest} object to be validated.
+     * @throws SAMLLogoutException If Signature of the Logout Request validation fails.
+     */
+    private void isValidLogoutReqSignature(LogoutRequest logoutRequest) throws SAMLLogoutException {
+
+        if (Boolean.parseBoolean(samlMessageContext.getFedIdPConfigs().get(IS_LOGOUT_REQ_SIGNED).toString()) &&
+                (!SAMLLogoutUtil.isValidSignature(logoutRequest, samlMessageContext))) {
```

Comment:
shall we extract this out to a private method? It would improve readability

Reply:
Fixed with PR #133